ts/src/delta.ts

```diff
@@ -2292,7 +2310,7 @@ export default class delta extends Exchange {
             direction = 'out';
         }
         type = this.parseLedgerEntryType (type);
-        const currencyId = this.safeInteger (item, 'asset_id');
+        const currencyId = this.safeString (item, 'asset_id');
```

Comment:
instead of adding `indexByStringifiedNumericId` you just need to replace the `this.safeValue (currenciesByNumericId, currencyId, currency)` with safeString no?

Reply:
No, there's another problem. Initial code uses `indexBy (this.currencies, 'numericId')` for setting `this.options.currenciesByNumericId`. As far as `numericId` is number, keys for `currenciesByNumericId` become numbers in py and strings in js and php. So in this code for py:
```
currencyId = self.safe_string(balance, 'asset_id')
currency = self.safe_value(currenciesByNumericId, currencyId)
```
`currency` is `None`

Reply:
@yzh-pelle got it, will check it